composer/utils/checkpoint.py

```diff
@@ -93,14 +96,13 @@ def _ensure_valid_checkpoint(checkpoint_filepath: Union[Path, str]) -> Union[Pat
 
     # No function name has been specified.
     if validate is None:
-        log.debug('No validation function specified. Skipping checkpoint validation.')
         return checkpoint_filepath
 
     # Validate the checkpoint.
-    if not validate(checkpoint_filepath):
-        raise ValueError(f'Checkpoint at {checkpoint_filepath} is invalid.')
+    if not validate(checkpoint_filepath, specs):
+        raise ValueError(f'Checkpoint at {checkpoint_filepath} {specs=}is invalid.')
```

Comment:
instead of printing out all the specs (could be quite long), can we adjust the return signature to return the (first) spec that failed (if applicable)?